sparsebit/quantization/observers/moving_average.py

```diff
@@ -9,15 +9,14 @@ class Observer(BaseObserver):
 
     def __init__(self, config, qdesc):
         super(Observer, self).__init__(config, qdesc)
+        assert hasattr(config.OBSERVER, "MOVING_AVERAGE") and self.qdesc.target == "feature", "Moving_average observer only support feature observing!"
         self.ema_ratio = config.OBSERVER.MOVING_AVERAGE.EMA_RATIO
 
     def calc_minmax(self):
-        assert (
-            self.qdesc.ch_axis > 0
-        ), "Moving_average observer only support feature observing!"
-        data = self.get_calibration_data(c_first=False)
+        data = self.data_cache.get_data_cache()
+        self.data_cache.reset()
         max_val, min_val = data[0].max(), data[0].min()
-        for i in range(1, data.shape[0]):
+        for i in range(1, len(data)):
```

Comment:
1. why from 1 not 0? 
2. why use len to get the batchsize of a tensor, not use size?